pinot-query-runtime/src/main/java/org/apache/pinot/query/mailbox/GrpcSendingMailbox.java

```diff
@@ -121,12 +112,31 @@ private void sendInternal(MseBlock block, List<DataBuffer> serializedStats)
     if (_contentObserver == null) {
       _contentObserver = getContentObserver();
     }
-    splitAndSend(block, serializedStats);
+    processAndSend(block, serializedStats);
     if (LOGGER.isDebugEnabled()) {
       LOGGER.debug("==[GRPC SEND]== message " + block + " sent to: " + _id);
     }
   }
 
+  private void processAndSend(MseBlock block, List<DataBuffer> serializedStats)
+      throws IOException {
+    _statMap.merge(MailboxSendOperator.StatKey.RAW_MESSAGES, 1);
+    long start = System.currentTimeMillis();
+    try {
+      DataBlock dataBlock = MseBlockSerializer.toDataBlock(block, serializedStats);
+      int sizeInBytes = _sender.processAndSend(dataBlock);
+      if (LOGGER.isDebugEnabled()) {
+        LOGGER.debug("Serialized block: {} to {} bytes", block, sizeInBytes);
```

Comment:
My two cents: If you know debug is enabled, as you do here, it is better to write the same log in the eager form:
```java
        LOGGER.debug("Serialized block: " + bytes + " to " + sizeInBytes + " bytes");
```

instead. Static analyzers like Intellij will complain, but they are wrong. The format you are using (varargs) requires an array allocation (which Java's JIT is very bad at optimizing) while my suggestion does not.

The varargs alternative has better performance than the eager one in cases you don't know for sure that the log will be printed because it is lazy and therefore the final string doesn't need to be built if the log is not printed.